### PR TITLE
Accuracy tweaks [ready]

### PR DESCRIPTION
**Pull request by Humonitarian**
## About The Pull Request

Removes point blank damage bonus, adds accuracy bonus instead.

Shooting while grabbed reduces accuracy.

Shooting someone grabbed increases accuracy (instead of damage).

Reduces chance to hit mobs close to the firer.

## Why It's Good For The Game

Melee buff, shotgun balance, more importance to recoil

## Changelog
:cl:
balance: aiming tweaks
balance: point blank damage bonus replaced with accuracy bonus
balance: grab gun damage bonus replaced with accuracy bonus (use headshots to kill)
/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -543,11 +549,28 @@
...
-				damage_mult = 2.5
+				accuracy_mult = 4
 			else if(grabstate >= GRAB_AGGRESSIVE)
-				damage_mult = 1.5
-	P.multiply_projectile_damage(damage_mult)
+				accuracy_mult = 8
+
+	P.multiply_projectile_accuracy(accuracy_mult)
+
+/obj/item/gun/proc/grabbed_inaccuracy(obj/item/projectile/P, mob/user) //TODO: make length of gun be a disadvantage
```
Comment (DiaFRAME444) [inline code/modules/projectiles/gun.dm:558]: Without the barrel length nerf, rifles will be weirdly good during CQC Grabs. In fact I'm pretty sure this makes point blanking someone's head off with a rifle better than using a proper melee weapon.

**Comment by DiaFRAME444** [comment]
"Shooting while grabbed reduces accuracy."
"Shooting someone grabbed increases accuracy (instead of damage)."
Overall this makes guns better at helping out grabfu than melee weapons. Even with the damage increase removed you'd be better finishing someone with a rifle than a NT halberd made for close quarters.

File: code/modules/projectiles/gun.dm (hunk 1 above)
Comment (Humonitarian) [reply code/modules/projectiles/gun.dm:558]: All guns are close to guaranteed to hit someone grabbed even before this buff. 
The buff itself is hardly significant however, the disadvantage on close range accuracy is identical to the advantage from aggressive grabs. (*4 increases leftmost bit by 3)
This is a pretty strong nerf to cqc-magdump as is, so holding off on merging just gives more power to magdumpers.

**Comment by Humonitarian** [comment]
> To continue with my earlier comment, "Shooting while grabbed reduces accuracy." "Shooting someone grabbed increases accuracy (instead of damage)." Overall this makes guns better at helping out grabfu than melee weapons. Even with the damage increase removed you'd be better finishing someone with a rifle than a NT halberd made for close quarters.

Same as with the review comment, this PR gives close to no accuracy bonus overall to grab-shooting, while significantly nerfing the actual bullets (easily less than half the damage!), and making guns less effective against a grab user.

**Comment by DiaFRAME444** [comment]
> (easily less than half the damage!),
Hold on, is this pointblank bonus 1.3 meaning +30% damage, or 130% damage?

**Comment by ErisMeleeEnjoyer** [comment]
> Without the barrel length nerf, rifles will be weirdly good during CQC Grabs. Without it, I'm pretty sure this makes point blanking someone's head off with a rifle better than using a proper melee weapon.

This makes pointblanking someone's head off impossible, as this greatly reduces damage(it stops being affected) with grabs. Isn't it also impossible to use a rifle with one hand? You use another to keep the grab, you know.

> Overall this makes guns better at helping out grabfu than melee weapons. Even with the damage increase removed you'd be better finishing someone with a rifle than a NT halberd made for close quarters.

How so? Melee weaponry is guaranteed to hit in close quarters if the player managed to click on the victim. It also might deal more damage, since, you know, no more ridicilous damage increase on grabs.

> Hold on, is this pointblank bonus 1.3 meaning +30% damage, or 130% damage?

Damage had a 1.3 multiplier for yellow/blue grabs, and a 2.5 multiplier for red grabs; which is +30% damage and 250% damage respectively. Now, think about it, does removing this damage increase and replacing it with accuracy make it better for grab masters?

**Comment by ValoTheValo** [comment]
Shotguns should keep PB. 

"Reduces chance to hit mobs close to the firer" also this just sounds really counterintuitive and dumb. If a guy or roach is up in my face he should be easier to hit not harder

File: code/modules/projectiles/projectile.dm [hunk 2]
```diff
@@ -299,6 +299,9 @@
 	if(target_mob == original)
 		var/acc_mod = leftmost_bit(projectile_accuracy)
 		hit_mod -= acc_mod //LOG2 on the projectile accuracy
+
+	// DEBUG FOR TEST
+	visible_message(SPAN_NOTICE("\The [src] had [(base_miss_chance[def_zone] + hit_mod) * 10]% chance to hit [target_mob]!"))
```
[the remark this answers is not in the document]
Comment (Humonitarian) [reply code/modules/projectiles/projectile.dm:304]: [TEST VERSION]

**Comment by ErisMeleeEnjoyer** [comment]
i literally said the same thing

**Comment by Humonitarian** [comment]
> Shotguns should keep PB.

They were buffed relatively, and buckshot has a lot more pellets to land

> "Reduces chance to hit mobs close to the firer" also this just sounds really counterintuitive and dumb. If a guy or roach is up in my face he should be easier to hit not harder

Yes, he's super easy to hit if you click on him, not behind him. Otherwise it's like if he was 5 tiles from you (still very close)

**Comment by ValoTheValo** [comment]
> > "Reduces chance to hit mobs close to the firer" also this just sounds really counterintuitive and dumb. If a guy or roach is up in my face he should be easier to hit not harder
> 
> Yes, he's super easy to hit if you click on him, not behind him. Otherwise it's like if he was 5 tiles from you (still very close)

you need to lead targets humon. still counterintuitive

**Comment by Humonitarian** [comment]
> > > "Reduces chance to hit mobs close to the firer" also this just sounds really counterintuitive and dumb. If a guy or roach is up in my face he should be easier to hit not harder
> > 
> > 
> > Yes, he's super easy to hit if you click on him, not behind him. Otherwise it's like if he was 5 tiles from you (still very close)
> 
> you need to lead targets humon. still counterintuitive

No you don't. It's point blank.

**Comment by Humonitarian** [comment]
Rebased onto #7499

File: code/modules/projectiles/projectile.dm [hunk 3]
```diff
@@ -299,6 +299,9 @@
 	if(target_mob == original)
 		var/acc_mod = leftmost_bit(projectile_accuracy)
 		hit_mod -= acc_mod //LOG2 on the projectile accuracy
+
+	// DEBUG FOR TEST
+	visible_message(SPAN_NOTICE("\The [src] had [(base_miss_chance[def_zone] + hit_mod) * 10]% chance to miss [target_mob]!"))
```
Comment (hyperioo) [inline code/modules/projectiles/projectile.dm:304]:
```suggestion
```
👁️

**Review by hyperioo — APPROVED** [review]
Update your stuff, 73 commits...

**Pull request merged**